tensorzero-internal/src/config_parser.rs

```diff
@@ -32,13 +36,60 @@ pub struct Config<'c> {
     pub templates: TemplateConfig<'c>,
 }
 
-#[derive(Debug, Default, PartialEq)]
+#[derive(Debug, Default)]
 pub struct GatewayConfig {
     pub bind_address: Option<std::net::SocketAddr>,
     pub observability: ObservabilityConfig,
+    pub object_store_data: ObjectStoreData,
     pub debug: bool,
 }
 
+#[derive(Clone, Debug, Default)]
+pub struct ObjectStoreData {
+    // TODO - should we instead make a dummy `ObjectStore` to represent an explicitly disabled store?
```

Comment:
we will probably want this soon -- if we're doing default fail -> explicitly specify we're gonna need to be able to distinguish unspecified from disabled